File: sdks/insuretech-go-sdk/pkg/models/underwriting_decision.go
package models

import (
	"time"
)

// UnderwritingDecision represents a underwriting_decision
type UnderwritingDecision struct {
	AdjustedPremium *Money `json:"adjusted_premium,omitempty"`
	AdjustmentReason string `json:"adjustment_reason,omitempty"`
	AuditInfo interface{} `json:"audit_info"`
	Conditions string `json:"conditions,omitempty"`
	DecidedAt time.Time `json:"decided_at"`
	Decision *DecisionType `json:"decision"`
	Id string `json:"id"`
	Method *DecisionMethod `json:"method"`
	PremiumAdjusted bool `json:"premium_adjusted,omitempty"`
	QuoteId string `json:"quote_id"`
	Reason string `json:"reason,omitempty"`
	RiskFactors string `json:"risk_factors,omitempty"`
	RiskLevel *UnderwritingRiskLevel `json:"risk_level,omitempty"`
	RiskScore string `json:"risk_score,omitempty"`
	UnderwriterComments string `json:"underwriter_comments,omitempty"`
	UnderwriterId string `json:"underwriter_id,omitempty"`
	ValidUntil time.Time `json:"valid_until,omitempty"`
}

File: backend/insurance_engine-old/src/InsuranceEngine.Underwriting/Domain/Entities/UnderwritingDecision.cs
using System;
using InsuranceEngine.Underwriting.Domain.Enums;
using InsuranceEngine.SharedKernel.Domain.ValueObjects;

namespace InsuranceEngine.Underwriting.Domain.Entities;

public class UnderwritingDecision
{
    public Guid Id { get; set; }
    public Guid QuoteId { get; set; }
    
    public DecisionType Decision { get; set; }
    public DecisionMethod Method { get; set; }

    public decimal RiskScore { get; set; }
    public RiskLevel RiskLevel { get; set; }
    public string? RiskFactorsJson { get; set; }

    public string? Reason { get; set; }
    public string? ConditionsJson { get; set; }

    public bool IsPremiumAdjusted { get; set; }
    public long AdjustedPremiumAmount { get; set; }
    public string AdjustedPremiumCurrency { get; set; } = "BDT";
    public string? AdjustmentReason { get; set; }

    public Guid? UnderwriterId { get; set; }
    public string? UnderwriterComments { get; set; }

    public DateTime DecidedAt { get; set; }
    public DateTime? ValidUntil { get; set; }

    public string? AuditInfoJson { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public Money AdjustedPremium => new(AdjustedPremiumAmount, AdjustedPremiumCurrency);
}
